subprojects/docs/src/docs/userguide/migration/upgrading_version_7.adoc

```diff
@@ -47,6 +47,19 @@ Instead, an Eclipse Temurin or IBM Semeru build is returned.
 Gradle 7.4+ will now emit a deprecation warning when the AdoptOpenJDK vendor is specified in the <<toolchains.adoc#sec:vendors,toolchain specification>> and it is used by auto provisioning.
 If you must use AdoptOpenJDK, you should turn off auto-download. If an Eclipse Temurin or IBM Semeru build works for you, specify `JvmVendorSpec.ADOPTIUM` or `JvmVendorSpec.IBM_SEMERU` as the vendor or leave the vendor unspecified.
 
+[[empty_directories_file_tree]]
+==== File trees and empty directory handling
+
+When using `@SkipWhenEmpty` on an input file collection, Gradle skips the task when it determines that the input is empty.
+If the input file collection consists only of file trees, Gradle ignores directories for the emptiness check.
+Though when checking for changes to the input file collection, Gradle only ignores directories when the `@IgnoreEmptyDirectories` annotation is present.
```

Comment:
I'd actually prefer keeping "file tree" and "file collection", since that is how we document those things: https://github.com/gradle/gradle/blob/04d8a8eb9c1a27d5fbbe4ec79789c9a69a91ba43/subprojects/docs/src/docs/userguide/authoring-builds/working_with_files.adoc#file-trees

Reply:
OK, I don't feel too strongly about this.